### ISSUE TITLE
`static` checking fails

### ISSUE THREAD

**Issue by tzturner** [techservicesillinois/awscli-login#104]
```bash
Run make static
mypy src/awscli_login/util.py src/awscli_login/const.py src/awscli_login/__init__.py src/awscli_login/config.py src/awscli_login/typing.py src/awscli_login/exceptions.py src/awscli_login/configure.py src/awscli_login/logger.py src/awscli_login/saml.py src/awscli_login/__main__.py setup.py src/tests/util.py src/tests/test_configure.py src/tests/base.py src/tests/__init__.py src/tests/test_util.py src/tests/exceptions.py src/tests/test_saml.py src/tests/cli/__init__.py src/tests/cli/test_logout.py src/tests/config/test_factor.py src/tests/config/util.py src/tests/config/test_profile.py src/tests/config/test_get_credentials.py src/tests/config/base.py src/tests/config/__init__.py src/tests/config/test_update.py  # Test only files that have been updated
src/tests/base.py:521: error: Too many arguments for "addCleanup" of "TestCase"
Found 1 error in 1 file (checked 2[7](https://github.com/techservicesillinois/awscli-login/runs/7240550167?check_suite_focus=true#step:8:8) source files)
make: *** [Makefile:125: .static] Error 1
Error: Process completed with exit code 2.
```

**Comment by ddriddle** [comment]
mypy indicates that `addCleanup` has too many arguments at the following line:

https://github.com/techservicesillinois/awscli-login/blob/53b18bb30bf06dc3e884c740253c51b4b2ae344a/src/tests/base.py#L521

But the online help for Python `3.9.0` indicates that pop can take two arguments:

```
Help on method pop in module collections.abc:

pop(key, default=<object object at 0x101318160>) method of os._Environ instance
    D.pop(k[,d]) -> v, remove specified key and return the corresponding value.
    If key is not found, d is returned if given, otherwise KeyError is raised.
```

We tried removing None from line 521 above, which made mypy happy, but it caused many fatal errors in our test suite. This is expected because if an environment variable is removed using pop that does not exist it will raise a KeyError as documented above. The reason None was passed originally to suppress the KeyError's during test setup.

In short, this bug is a mypy bug. We should followup with a bug report to mypy.
